app/src/androidTest/java/org/sagebionetworks/research/mpower/FromEntryActivityToSettingsTest.kt

```diff
@@ -0,0 +1,296 @@
+ * other materials provided with the distribution.
+ *
+ * 3.  Neither the name of the copyright holder(s) nor the names of any contributors
+ * may be used to endorse or promote products derived from this software without
+ * specific prior written permission. No license is granted to the trademarks of
+ * the copyright holders even if such marks are included in this software.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package org.sagebionetworks.research.mpower
+
+import android.view.View
+import android.view.ViewGroup
+import androidx.test.espresso.Espresso
+import androidx.test.espresso.Espresso.onView
+import androidx.test.espresso.NoMatchingViewException
+import androidx.test.espresso.action.ViewActions
+import androidx.test.espresso.assertion.ViewAssertions.matches
+import androidx.test.espresso.matcher.ViewMatchers
+import androidx.test.espresso.matcher.ViewMatchers.withText
+import androidx.test.filters.LargeTest
+import androidx.test.rule.ActivityTestRule
+import androidx.test.runner.AndroidJUnit4
+import it.xabaras.android.espresso.recyclerviewchildactions.RecyclerViewChildActions.Companion.childOfViewAtPositionWithMatcher
+import org.hamcrest.Description
+import org.hamcrest.Matcher
+import org.hamcrest.Matchers
+import org.hamcrest.TypeSafeMatcher
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+
+@LargeTest
+@RunWith(AndroidJUnit4::class)
+class FromEntryActivityToSettingsTest {
+
+    @Rule
+    @JvmField
+    var mActivityTestRule = ActivityTestRule(EntryActivity::class.java)
+    var skipSingInPart = false
+
+    @Before
+    fun setupTest() {
+        try {
+            //check that user is not signed in. If he is, skip sing in part
+            val appCompatButton = Espresso.onView(
+                    Matchers.allOf(ViewMatchers.withId(R.id.button_go_forward), ViewMatchers.withText("Next"),
+                            childAtPosition(
+                                    childAtPosition(
+                                            ViewMatchers.withId(R.id.bp_next_button_container),
+                                            0),
+                                    2),
```

Comment:
Is there a way to reference this element without using a hard coded position of `2`?  It would be cool if this test were tolerant to UI redesigns.

Reply:
This is the reason why 2 is here. Make sure that you did not move this button because of some code changed. But I can delete it in this place, it's not necessary to check

Reply:
Just to make sure that i understand: those `childAtPosition` calls just make sure that no one has changed the layout and moved the button around, correct?  In that case, i think that we're better off without them if we're still matching on the id and/or the text in the button.

Does that make sense?